src/journal-remote/journal-gatewayd.c

```diff
@@ -181,11 +183,12 @@ static ssize_t request_reader_entries(
                 } else if (r == 0) {
 
                         if (m->follow) {
-                                r = sd_journal_wait(m->journal, (uint64_t) -1);
+                                r = sd_journal_wait(m->journal, (uint64_t) JOURNAL_WAIT_TIMEOUT);
                                 if (r < 0) {
                                         log_error_errno(r, "Couldn't wait for journal event: %m");
                                         return MHD_CONTENT_READER_END_WITH_ERROR;
-                                }
+                                } else if (r == SD_JOURNAL_NOP)
```

Comment:
My objection was about the `else`, not about the `if`. If the branch above is executed, it will `return` in any case, so the `else` is pointless.